### ISSUE TITLE
Failing behat tests still allow travisci to "pass"

### ISSUE THREAD

**Issue by arknoll** [acquia/blt#1375]
My system information:
* Operating system type: TravisCI
* BLT version: 8.8.1

When I run this command:
```
source $BLT_DIR/scripts/travis/run_tests
```

I get the following output:
Behat tests fail. However, build continues as if it passed.

And I expected this to happen:
travisci to fail build.

**Comment by arknoll** [comment]
See https://travis-ci.com/acquia-pso/canary/builds/43994635

**Comment by geerlingguy** [comment]
Yeah, I was doing some local troubleshooting and found the same thing:

```
    features/Examples.feature:9

4 scenarios (3 passed, 1 failed)
29 steps (28 passed, 1 failed)
0m41.50s (76.08Mb)
[Testing\Behat]  Exit code 1 
[error]  Behat tests failed 

 13:15:33 ~/Sites/acquia/canary $ echo $?
0
```

Basically, it looks like Robo is not passing through the exit code (even though it obviously knows the exit code was `1`).

**Comment by geerlingguy** [comment]
It looks like the task should return a `Robo/Result`: http://robo.li/getting-started/#result

**Comment by geerlingguy** [comment]
Strangely, it looks like the exit code is not being passed back inside `Acquia\Blt\Robo`'s `run()` method:

```
// Always returns $status_code of '0'.
$status_code = $this->runner->run($input, $output, NULL, $this->commands);
```

**Comment by grasmash** [comment]
I suspect that this is due to the docker execution call. I'm not able to reproduce this issue without Docker.

**Comment by grasmash** [comment]
Hm. I verified that when blt-run-robo.php executes `exit($status_code);`, `$status_code` is actually 1.

**Comment by geerlingguy** [comment]
I'm going to see if your fix here gets the command to pass: https://github.com/acquia/blt/pull/1381/files

**Comment by geerlingguy** [comment]
#1377 is the correct fix for this.

**Comment by grasmash** [comment]
Fixed by #1381
